network/dispatch.go

```diff
@@ -0,0 +1,129 @@
+package network
+
+import (
+	"errors"
+	"sync"
+)
+
+// Dispatcher is an interface whose sole role is to distribute messages to the
+// right Processor. No processing is done,i.e. no looking at packet content.
+// There are many ways to distribute messages: for the moment, only
+// BlockingDispatcher is implemented, which is a blocking dispatcher.
+// Later, one can easily imagine to have a dispatcher with one worker in a
+// goroutine or a fully fledged producer/consumers pattern in go routines.
+// Each Processor that wants to receive all messages of a specific
+// type must register itself to the dispatcher using `RegisterProcessor()`.
+// The network layer must call `Dispatch()` each time it receives a message, so
+// the dispatcher is able to dispatch correctly to the right Processor for
+// further analysis.
+type Dispatcher interface {
+	// RegisterProcessor is called by a Processor so it can receive all packets
```

Comment:
No no, ex:
```
d := NewBlockingDispatcher()
p := MyProcessor()
d.RegisterProcessor(MyMessageType, p)
```
Ok while writing I see what you mean, but saying that the Dispatcher calls the method is also ambiguous (because the dispatcher does not know the processor necessarily, usually, it's the processor who has access to the Dispatcher so it can register itself to it) so I'll just remove that part of the comment.